app/src/main/java/org/cimbar/camerafilecopy/MainActivity.java

```diff
@@ -158,7 +159,24 @@ public Mat onCameraFrame(CvCameraViewFrame frame) {
         String res = processImageJNI(mat.getNativeObjAddr(), this.dataPath, this.modeVal);
 
         // res will contain a file path if we completed a transfer. Ask the user where to save it
-        if (!res.isEmpty()) {
+        if (res.startsWith("/")) {
+            if (res.length() >= 2 && res.charAt(1) == '4') {
+                detectedMode = 4;
+                mModeSwitch.setActivated(true);
```

Comment:
I may regret this...

Reply:
Decoded to move it onto the UI thread. It's still a hack, but it feels slightly less bad now?